retrieval_service/main.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import gitlab
+from time import sleep
+import os
+from prometheus_client import Counter, Histogram
+
+from common.constants import (
+    project_events,
+    user_events,
+    issue_events,
+    issue_note_events,
+    group_events,
+    event_types,
+)
+
+from common.event_processor import EventProcessor
+
+LOGLEVEL = os.environ.get('LOGLEVEL', 'WARNING').upper()
+logging.basicConfig(
+    level=LOGLEVEL, format="%(asctime)s - %(levelname)s - Retrieval service: %(message)s"
+)
+
+# GitlabRetrievalProcessor class is used to process events from Redis queues
+# and push events back into to Redis queues after processing.
+# It is a subclass of EventProcessor.
+# It is used to retrieve data from GitLab using the GitLab API.
+class GitlabRetrievalProcessor(EventProcessor):
+    def __init__(self, GITLAB_URL, GITLAB_ACCESS_TOKEN, redis_conn=None, testing=False):
+        super().__init__("verification", event_types, redis_conn)
+        self.gitlab_client = gitlab.Gitlab(
+            GITLAB_URL, private_token=GITLAB_ACCESS_TOKEN
+        )
+        self.testing = testing
+
+        self.event_processing_time = Histogram(
+            "retrieval_service_event_processing_seconds",
+            "Time taken to process an event",
+        )
+        self.events_processed = Counter(
+            "retrieval_service_events_processed_total",
+            "Total number of events processed",
+        )
+
+    def process_event(self, queue_name, event_data):
+        with self.event_processing_time.time():
+            event_type = queue_name.split("_", 1)[-1]
+
+            # Determine how to retrieve data based on event type
+            if event_type in user_events:
+                gitlab_object = self._process_user_event(event_data)
+            elif event_type in project_events:
+                gitlab_object = self._process_project_event(event_data)
+            elif event_type in issue_events:
+                gitlab_object = self._process_issue_event(event_data)
+            elif event_type in issue_note_events:
+                gitlab_object = self._process_issue_note_event(event_data)
+            elif event_type in group_events:
+                gitlab_object = self._process_group_event(event_data)
+            else:
+                logging.info(f"{self.__class__.__name__}: event {event_type} received")
+                return
+
+            self.events_processed.inc()
```

Comment:
**question:** should this be incremented here? If gitlab_object is none, do we want the event to be counted as processed?